README.md

```diff
@@ -211,6 +217,16 @@ The proposal here is to make the `document.domain` setter do nothing (perhaps em
 
 However, throwing would pose a potential migration burden for sites. Measurements from Chrome show that the `document.domain` setter is widely used ([~12% of page views](https://chromestatus.com/metrics/feature/timeline/popularity/739)), but only has an effect a smaller percentage of the time ([~0.28%](https://chromestatus.com/metrics/feature/timeline/popularity/2543) + [~0.52%](https://chromestatus.com/metrics/feature/timeline/popularity/2544) of page views). If we made it throw, then all of that 12% would need to update their code before they could take advantage of origin isolation. By making it do nothing, we instead confine the migration cost to the <1%.
 
+### `window.originIsolationRestricted`
+
+The `window.originIsolationRestricted` getter can feel a bit redundant. After all, the web developer should know whether origin isolation applies: they set the header themselves!
+
+However, as explained [above](#specification-plan), the `Origin-Isolation` header cannot always be respected. If a web developer has configured their server inconsistently, or if they are in the middle of changing their server configuration, other same-origin documents in the browsing context group might have different values for the header, and the overriding consistency requirement would then take over. Thus, `window.originIsolationRestricted` gives runtime insight into whether isolation succeeded, which can be useful for reporting, or for detecting misconfigurations.
+
+Finally, there are some subtleties around what this getter should return in various edge cases. As indicated by the name, it's currently designed to return true only if origin isolation causes _restrictions_ to occur. This means that it doesn't exist in workers, where the concept isn't applicable. And this means that it returns `false` for pages with opaque origins, or whose origin is an IP address or is in the public suffix list, because for those cases [the page's site is the same as its origin](https://html.spec.whatwg.org/#obtain-a-site).
```

Comment:
I think the claim for workers is a bit too strong and it might still make sense to expose this in dedicated workers in the future. A dedicated worker could have a message channel with a cross-origin-same-site document or dedicated worker, for instance, and use that to share WebAssembly modules. Unlikely and an edge case, but nevertheless valid.

Reply:
I think this is covered by https://github.com/WICG/origin-isolation#more-detail-on-workers:

> A worker could send a WebAssembly.Module to a same-site cross-origin destination by first passing it (or a MessageChannel through which the WebAssembly.Module gets transmitted) to a same-origin document, and letting the document try to go beyond the origin boundary. This would no longer be possible with this proposal, **but only because of the proposal's effect on the document, not because of any effect it had on the worker**.

Reply:
Fair enough.